src/rules/no_eval.rs

```diff
@@ -59,23 +63,61 @@ impl<'c> NoEvalVisitor<'c> {
   fn new(context: &'c mut Context) -> Self {
     Self { context }
   }
+
+  fn maybe_add_diagnostic(&mut self, source: &dyn StringRepr, span: Span) {
+    if source.string_repr().unwrap() == "eval" {
+      self.add_diagnostic(span);
+    }
+  }
+
+  fn add_diagnostic(&mut self, span: Span) {
+    self.context.add_diagnostic_with_hint(
+      span,
+      "no-eval",
+      "`eval` call is not allowed",
+      "Remove the use of `eval`",
+    );
+  }
+
+  fn handle_paren_callee(&mut self, p: &ParenExpr) {
+    match p.expr.as_ref() {
+      // Nested paren callee ((eval))('var foo = 0;')
+      Expr::Paren(paren) => self.handle_paren_callee(paren),
+      // Single argument callee: (eval)('var foo = 0;')
+      Expr::Ident(ident) => self.maybe_add_diagnostic(ident, ident.span),
+      // Multiple arguments callee: (0, eval)('var foo = 0;')
+      Expr::Seq(seq) => {
+        for expr in &seq.exprs {
+          if let Expr::Ident(ident) = expr.as_ref() {
+            self.maybe_add_diagnostic(ident, ident.span)
+          }
+        }
+      }
+      _ => {}
+    }
+  }
 }
 
 impl<'c> Visit for NoEvalVisitor<'c> {
   noop_visit_type!();
 
+  fn visit_var_declarator(&mut self, v: &VarDeclarator, _: &dyn Node) {
+    if let Some(expr) = &v.init {
+      if let Expr::Ident(ident) = expr.as_ref() {
+        self.maybe_add_diagnostic(ident, v.span);
+      }
+    }
+  }
+
   fn visit_call_expr(&mut self, call_expr: &CallExpr, _parent: &dyn Node) {
     if let ExprOrSuper::Expr(expr) = &call_expr.callee {
-      if let Expr::Ident(ident) = expr.as_ref() {
-        let name = ident.sym.as_ref();
-        if name == "eval" {
-          self.context.add_diagnostic_with_hint(
-            call_expr.span,
-            "no-eval",
-            "`eval` call is not allowed",
-            "Remove the use of `eval`",
-          );
+      match expr.as_ref() {
+        Expr::Ident(ident) => self.maybe_add_diagnostic(ident, call_expr.span),
+        Expr::Member(member_expr) => {
+          self.maybe_add_diagnostic(member_expr, call_expr.span)
         }
```

Comment:
This could result in false positive. For example:

```ts
const obj = {
  eval: (val) => { console.log(val); },
};

obj.eval("yeah");
```

In this case calling `obj.eval` is totally fine, but your code will report it mistakenly.